tests/functional/rbac/test_permissions_deleting.py

```diff
@@ -0,0 +1,152 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test cases of permissions deleting"""
+from adcm_client.objects import ADCMClient
+
+from tests.functional.rbac.conftest import BusinessRoles, create_policy, is_allowed, is_denied
+
+
+def test_remove_user_from_policy(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if user removed from policy user list
+    """
+    cluster_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    policy = create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(user=[{"id": sdk_client_fs.user(username="admin").id}])
+    is_denied(cluster, BusinessRoles.ViewConfigurations)
+
+
+def test_remove_group_from_policy(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if group with user removed from policy group list
+    """
+    cluster_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    group = sdk_client_fs.group_create("test_group", user=[{"id": user.id}])
+    empty_group = sdk_client_fs.group_create("empty_group")
+    policy = create_policy(
+        sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[group]
+    )
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(group=[{"id": empty_group.id}])
+    is_denied(cluster, BusinessRoles.ViewConfigurations)
+
+
+def test_remove_user_from_group(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if user removed from group with policy
+    """
+    cluster_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    group = sdk_client_fs.group_create("test_group", user=[{"id": user.id}])
+    create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[group])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    group.update(user=[])
+    is_denied(cluster, BusinessRoles.ViewConfigurations)
+
+
+def test_remove_object_from_policy(user_sdk: ADCMClient, user, prepare_objects, sdk_client_fs):
+    """
+    Test that user loses access if object changed from policy
+    """
+    cluster_via_admin, service_via_admin, *_ = prepare_objects
+    cluster = user_sdk.cluster(id=cluster_via_admin.id)
+    policy = create_policy(sdk_client_fs, BusinessRoles.ViewConfigurations, objects=[cluster], users=[user], groups=[])
+    is_allowed(cluster, BusinessRoles.ViewConfigurations)
+    policy.update(object=[{"id": service_via_admin.id, "type": "service"}])
+    is_denied(cluster, BusinessRoles.ViewConfigurations)
```

Comment:
Maybe add a check that view service config is now available? It looks like possible problem with reassignment of permissions

Reply:
fixed